File: 20221207/Expression/src/test/java/ExpressionTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ExpressionTest {
    @Test
    void solution() {
        Expression test = new Expression();

        assertEquals(4, test.solution(15));
    }

    @Test
    void validate() {
        Expression test = new Expression();

        assertTrue(test.validate(1, 15));
        assertTrue(test.validate(4, 15));
        assertTrue(test.validate(7, 15));
        assertTrue(test.validate(15, 15));
        assertFalse(test.validate(2, 15));
        assertFalse(test.validate(3, 15));
    }
}

File: 20221208/Expression.test.js
import solution, { validate } from './Expression';

test('solution', () => {
  expect(solution(15)).toBe(4);
});

test('validate', () => {
  expect(validate(1, 15)).toBe(true);
  expect(validate(4, 15)).toBe(true);
  expect(validate(7, 15)).toBe(true);
  expect(validate(15, 15)).toBe(true);
  expect(validate(2, 15)).toBe(false);
  expect(validate(3, 15)).toBe(false);
});
